### PR TITLE
48522 unicode tests

### PR DESCRIPTION
**Pull request by ricellis**
*What*
Refactor the unicode tests for the new HTTP API.

*How*
Replace usage of the Apache HTTP classes in the test methods.
Fix print lines with logging.
Update the test to use the new JUnit rules.

*Testing*
The tests pass after the refactor.

reviewer @emlaver 
reviewer @tomblench

### CODE REVIEW

File: src/test/java/com/cloudant/tests/UnicodeTest.java [hunk 1]
```diff
@@ -94,21 +83,26 @@ public void tearDown() {
...
         // -the-character-set-of-the-page-with-a-head
-        ContentType contentType;
-        try {
-            contentType = ContentType.getOrDefault(entity);
-        } catch (UnsupportedCharsetException e) {
-            throw new RuntimeException("The output of " + uri + " is in charset " + e
-                    .getCharsetName() + ", which is unsupported.",
-                    e);
+
+        String contentType = connection.getConnection().getHeaderField("Content-Type");
```
Comment (emlaver) [inline src/test/java/com/cloudant/tests/UnicodeTest.java]: Can we not use `connection.getConnection().getContentType()`?

Comment (emlaver) [inline src/test/java/com/cloudant/tests/UnicodeTest.java]: There's also `connection.getConnection().getContentEncoding()`.

Comment (ricellis) [inline src/test/java/com/cloudant/tests/UnicodeTest.java]: Yes, `getContentType()` refers to the remote object so we can use that - fixed in 3b5db94.
`getContentEncoding()` does not get charset information for text types, it is used for entire body compression schmes (e.g. `gzip`) so we can't use that one.

**Comment by tomblench** [comment]
+1

**Comment by emlaver** [comment]
Looks good, +1

**Pull request merged**